app/controllers/events/view_transcript.py

```diff
@@ -0,0 +1,72 @@
+from app.models.course import Course
+from app.models.courseParticipant import CourseParticipant
+from app.models.program import Program
+from app.models.courseInstructor import CourseInstructor
+from app.models.user import User
+from app.models.term import Term
+from app.models.eventParticipant import EventParticipant
+from app.models.event import Event
+from peewee import DoesNotExist
+
+from peewee import *
+
+def getSlCourseTranscript(user):
+    """
+    Returns a service-learning course list that contains a name of the course, term,
+    list of instructors who teach the course, and hours earned for each course that the user took.
+    :user: model object
+    """
+    username = User.get_by_id(user)
+    slCourseInformation = (CourseParticipant
+    .select(CourseParticipant.course, CourseParticipant.user, fn.SUM(CourseParticipant.hoursEarned).alias("hoursEarned"))
+    .group_by(CourseParticipant.course, CourseParticipant.user)
+    .where(CourseParticipant.user == username))
+
+    allCoursesList = []
+    listOfEachSlCourseInfo = []
+    
+    for course in slCourseInformation:
+        user_full_name = course.user.firstName + " "+ course.user.lastName
+        course_name = course.course.courseName
+        description = course.course.term.description
+        hours = course.hoursEarned
+        cInstructor = CourseInstructor.select().where(CourseInstructor.course== course.course)
+        instructorList = []
+        for instructor in cInstructor.objects():
+            instructorList.append(instructor.user.firstName+" "+ instructor.user.lastName)
+        #creates a list for all information of a course that a courseParticipant is involved in

```

Comment:
- Because this select is a simple select, you don't need to use `.objects()` to get the values out (I'm pretty sure). 
- `cInstructor` isn't a great name, because it is potentially a list, so it should be plural or, better yet, an indication that it is a query object. 
- When you have a simple transform of a list in python, writing it as a list comprehension is preferred. 
- Another consideration is that you don't need to do that javascript-like ` + " " + ` string concatentation because of f-strings in python. For short strings like this one the readability is a matter of personal preference, so that's up to you if you want to change it. 

Taking all that together, this code might be something like:

`instructorQuery = CourseInstructor.select().where(CourseInstructor.course== course.course)`
`instructorList = [f"{i.user.firstName} {i.user.lastName}" for i in instructorQuery]`